<launch>
  <arg name="image" />
  <arg name="camera" default="driver" />
  <arg name="autostart" default="True" />
  <arg name="num_cols" default="8" />
  <arg name="num_rows" default="6" />
  <arg name="dim" default="1" />
  <arg name="pattern" default="0" />
  <arg name="flags" default="0" />
  <arg name="use_service" default="True" />
  <arg name="features" default="features" />
  <arg name="camera_info" default="camera_info" />
  <arg name="camera_name" default="camera" />
  <arg name="file_name" default="camera.yaml" />


  <include file="$(find image_cb_detector)/launch/image_cb_detector.launch">
    <arg name="image" value="$(arg image)" />
    <arg name="scaling" value="0.25" />
    <arg name="autostart" value="$(arg autostart)" />
    <arg name="num_rows" value="$(arg num_rows)" />
    <arg name="num_cols" value="$(arg num_cols)" />
    <arg name="dim" value="$(arg dim)" />
    <arg name="pattern" value="$(arg pattern)" />
    <arg name="flags" value="$(arg flags)" />
  </include>

  <node type="mono_calibrator.py"
      pkg="camera_intrinsic_calibrator"
      name="mono_calibrator"
      args="--autostart $(arg autostart) --use_service $(arg use_service) --camera_name $(arg camera_name) --file_name $(arg file_name)"
      output="screen"
      required="true">
      <remap from="autostart" to="autostart" />
      <remap from="camera" to="$(arg camera)" />
      <remap from="features" to="$(arg features)" />
      <remap from="camera_info" to="$(arg camera_info)" />
  </node>
</launch>
